client/allocrunner/taskrunner/task_runner_hooks.go

```diff
@@ -90,15 +90,22 @@ func (tr *TaskRunner) initHooks() {
 
 	// If Vault is enabled, add the hook
 	if task.Vault != nil {
+		authMethod := vaultAuthMethodToken
+		if wid := task.GetIdentity("vault"); wid != nil {
+			authMethod = vaultAuthMethodJWT
+		}
```

Comment:
The hook has access to the allocation, and therefore the task. It might be nice from the standpoint of encapsulation to push this method switching down into the hook rather than surfacing it in the hook config. But I don't feel strongly about it, especially if it's a lot of extra work to do.

Reply:
Moved logic to `newVaultHook`. I'm now passing the whole task instead of just the task name.